src/Features/VisualBasic/Portable/ChangeSignature/VisualBasicChangeSignatureService.vb

```diff
@@ -429,124 +444,169 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ChangeSignature
             Return vbnode
         End Function
 
+        Private Function IsParamsArrayExpanded(semanticModel As SemanticModel, node As SyntaxNode, symbolInfo As SymbolInfo, cancellationToken As CancellationToken) As Boolean
+            If symbolInfo.Symbol Is Nothing Then
+                Return False
+            End If
+
+            Dim argumentCount As Integer
+            Dim lastArgumentIsNamed As Boolean
+            Dim lastArgumentExpression As ExpressionSyntax
+
+            Dim invocation = TryCast(node, InvocationExpressionSyntax)
+            Dim objectCreation = TryCast(node, ObjectCreationExpressionSyntax)
+            If invocation IsNot Nothing Then
+                argumentCount = invocation.ArgumentList.Arguments.Count
+                Dim isNamed = invocation.ArgumentList.Arguments.LastOrDefault()?.IsNamed
+                lastArgumentIsNamed = isNamed.HasValue AndAlso isNamed.Value
+                lastArgumentExpression = invocation.ArgumentList.Arguments.LastOrDefault()?.GetExpression()
+            ElseIf objectCreation IsNot Nothing Then
+                argumentCount = objectCreation.ArgumentList.Arguments.Count
+                Dim isNamed = objectCreation.ArgumentList.Arguments.LastOrDefault()?.IsNamed
+                lastArgumentIsNamed = isNamed.HasValue AndAlso isNamed.Value
+                lastArgumentExpression = objectCreation.ArgumentList.Arguments.LastOrDefault()?.GetExpression()
+            Else
+                Throw New ArgumentException("Unexpected SyntaxNode", NameOf(node))
+            End If
+
+            Dim methodSymbol = TryCast(symbolInfo.Symbol, IMethodSymbol)
+            If methodSymbol IsNot Nothing AndAlso methodSymbol.Parameters.LastOrDefault()?.IsParams Then
+                If argumentCount > DirectCast(symbolInfo.Symbol, IMethodSymbol).Parameters.Length Then
+                    Return True
+                ElseIf argumentCount = DirectCast(symbolInfo.Symbol, IMethodSymbol).Parameters.Length Then
+                    If lastArgumentIsNamed Then
+                        Return True
+                    Else
+                        Dim fromType = semanticModel.GetTypeInfo(lastArgumentExpression, cancellationToken)
+                        Dim toType = DirectCast(symbolInfo.Symbol, IMethodSymbol).Parameters.Last().Type
+                        Return Not semanticModel.Compilation.HasImplicitConversion(fromType.Type, toType)
+                    End If
+                Else
+                    Return False
+                End If
+            End If
+
+            Return False
+        End Function
```

Comment:
is this the same as the C# version?  it seems familiar.

Reply:
Deduped most of this logic (everything except the initial data gathering).